packages/shared/src/components/feeds/FeedContainer.tsx

```diff
@@ -167,7 +168,10 @@ export const FeedContainer = ({
   const cardContainerStyle = { ...getStyle(isList, spaciness) };
   const isFinder = router.pathname === '/search/posts';
   const isSearch = showSearch && !isFinder;
-  const customFeedPlacement = useFeature(featureCustomFeedPlacement);
+  const { value: customFeedPlacement } = useConditionalFeature({
+    feature: featureCustomFeedPlacement,
+    shouldEvaluate: !insaneMode && isLaptop,
+  });
```

Comment:
Although we have the `insaneMode` in the condition already, we need another variable to hold the actual evaluation to handle the edge cases.

For example, the user was on grid mode, hence enrolled, but toggled the insane mode. The value of `customFeedPlacement` will always be `true` now and will not change since the user was already enrolled.

You will have to do something similar to below example:
```suggestion
  const shouldEvaluate = !insaneMode && isLaptop;
  const { value: customFeedPlacement } = useConditionalFeature({
    feature: featureCustomFeedPlacement,
    shouldEvaluate,
  });
  const shouldShow = customFeedPlacement && shouldShow
```

Reply:
Maybe using a hook would be ideal so we won't have to do this multiple consts everywhere.

Reply:
That's a great observation. I'll update it